akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/BehaviorTestKitImpl.scala

```diff
@@ -30,7 +30,7 @@ private[akka] final class BehaviorTestKitImpl[T](_path: ActorPath, _initialBehav
     with akka.actor.testkit.typed.scaladsl.BehaviorTestKit[T] {
 
   // really this should be private, make so when we port out tests that need it
-  private[akka] val context = new EffectfulActorContext[T](_path, () => currentBehavior)
+  private[akka] val context: EffectfulActorContext[T] = new EffectfulActorContext[T](_path, () => currentBehavior)
```

Comment:
`context` overrides, and Scala 3 by default uses the same type as in the superclass. The explicit type is required for giving the override a more specific type (`EffectfulActorContext` instead of `TypedActorContext`)